src/Core/Children.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @type {Object}
+ */
+var Children = {
+	/**
+	 * @param {*} children 
+	 * @return {Array}
+	 */
+	toArray: function toArray (children) {
+		var array = []
+
+		if (children == null)
+			return array
+		else if (isValidElement(children) || typeof children !== 'object')
+			return [children]
+		else if (children instanceof Array)
+			array = children
```

Comment:
by not returning here, you're supporting `[[child]]` which seems unintended?

Reply:
This mimics React behavior, in this case `toArray` always returns a flat array.

Reply:
I see, and flatten is deep.
I'd prefer a stricter implementation, both for consistency of interface and performance, but I understand following React

Reply:
Always creating a flat array allows us to sort the children, for example we could have a component `<Foo>` that we pass a set of children from different data sources.

```js
<Foo><h1></h1>{array.map(x=><span>x<span>)}<p></p>{data.map(x=><a>x<a>)}<Foo>
```
This would produce `[h1, [spans], p, [anchors]]`. If we wanted to sort the children before rendering them in the `<Foo>` component a flat would make this easier to work with array.

Reply:
I see. Personally I tend to use `concat()`, but I understand allowing a looser interface
```
[h1].concat([spans].sort()).concat([p]).concat([anchors].sort()).concat([a])
```